### PR TITLE
look up the patch data using a patch data index

### PR DESCRIPTION
**Pull request by boyceg**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by drwells** [comment]
Ooops. Yet case of "pointers can be converted to integers".

What's a bit surprising about this one is that I thought the new tests in that PR would execute that code path - I'll fix the test and append that to this PR.

File: ibtk/src/math/HierarchyMathOps.cpp [hunk 1]
```diff
@@ -1987,7 +1987,7 @@ HierarchyMathOps::interp(const int dst_idx,
                          const bool src_cf_bdry_synch)
 {
     if (src_ghost_fill) src_ghost_fill->fillData(src_ghost_fill_time);
-    Pointer<NodeDataFactory<NDIM, double> > data_factory = dst_var->getPatchDataFactory();
+    Pointer<CellDataFactory<NDIM, double> > data_factory = dst_var->getPatchDataFactory();
```
Comment (drwells) [inline ibtk/src/math/HierarchyMathOps.cpp:1990]: I fixed another copy-paste error.

**Review by drwells — APPROVED** [review]

**Comment by drwells** [comment]
While I was at it - I noticed that the existing nodal interpolation tests weren't wrong, but they used periodic BCs and a nonperiodic manufactured solution which caused the infinity norm to be O(1). I fixed that and added a new test for node-to-cell.

File: ibtk/src/math/HierarchyMathOps.cpp (hunk 1 above)
Comment (boyceg) [reply ibtk/src/math/HierarchyMathOps.cpp:1990]: Gotta get rid of the silently failing implicit conversion facilitated by `Pointer`.

**Pull request merged**